### PR TITLE
Fixes #21952: opt in to jquery select plugins.

### PR DESCRIPTION
**Pull request by waldenraines**
Instead of assuming all selects should use jquery select plugins
we should instead use an opt in model.

http://projects.theforeman.org/issues/21952

### CODE REVIEW

File: app/helpers/form_helper.rb [hunk 1]
```diff
@@ -172,6 +172,13 @@ def time_zone_select_f(f, attr, default_timezone, select_options = {}, html_opti
 
   def selectable_f(f, attr, array, select_options = {}, html_options = {})
     html_options[:size] = 'col-md-10' if html_options[:multiple]
+
+    unless html_options[:disable_select2] do
```
[the remark this answers is not in the document]
Comment (waldenraines) [reply app/helpers/form_helper.rb]: Oops, thanks @houndci-bot

**Comment by waldenraines** [comment]
[test foreman]

**Review by dLobatog — COMMENTED** [review]
@waldenraines Why shouldn't it remain opt-out? I remember the first time we did a demo to Patternfly folks and @rohoover they praised them - from an UX point of view I use them all the time.

**Comment by waldenraines** [comment]
> @waldenraines Why shouldn't it remain opt-out? I remember the first time we did a demo to Patternfly folks and @Rohoover they praised them - from an UX point of view I use them all the time.

It will still be opt-out from the rails perspective (as long as you use a method in `form_helper.rb`) but it is bad practice to assume all `<select>`s should be using this jquery library.  This actively breaks every new select in react.

There is no pattern for this in patternfly either.

**Comment by waldenraines** [comment]
[test katello]

**Comment by waldenraines** [comment]
[test foreman]

**Comment by waldenraines** [comment]
[test]

File: app/helpers/form_helper.rb [hunk 2]
```diff
@@ -67,10 +67,8 @@ def multiple_selects(f, attr, associations, selected_ids, options = {}, html_opt
     end
     authorized = AssociationAuthorizer.authorized_associations(associations.reorder(nil), klass).all
 
-    # select2.js breaks the multiselects disabled items location
-    # http://projects.theforeman.org/issues/12028
     html_options["class"] ||= ""
-    html_options["class"] += " without_select2"
+    html_options["class"] += " jquery-multiselect"
```
Comment (ohadlevy) [inline app/helpers/form_helper.rb:71]: not sure I understand? without_select2 class avoids using select2, I was under the assumption you want to move from a black list (when someone adds the without_select2) to a whitelist of when we use select2?

**Comment by ohadlevy** [comment]
@waldenraines did you see https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/forms/Select.js ?

File: app/helpers/form_helper.rb (hunk 2 above)
Comment (waldenraines) [reply app/helpers/form_helper.rb:71]: Yes, but I wanted to keep the form generating code the same so as to minimize changes in the application.  Basically I just want normal selects that I add with react to not automatically get the select2/multiselect classes.

**Comment by waldenraines** [comment]
> @waldenraines did you see https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/components/common/forms/Select.js ?

But wouldn't that still get both the problematic selects applied to it?

**Comment by waldenraines** [comment]
[test]

**Comment by waldenraines** [comment]
> [test]

I think these tests are likely failing in develop as well because they expect an id of `#s2id_host_hostgroup_id` and the id in the application itself is `#s2d_autogen3` unless there is some other magic that I'm not seeing here.  

@dLobatog @ohadlevy thoughts?

**Comment by waldenraines** [comment]
> @dLobatog @ohadlevy thoughts?

@dLobatog @ohadlevy ping

**Comment by beav** [comment]
this patch fixed an issue I hit when testing another PR

**Comment by waldenraines** [comment]
Closing in favor of https://github.com/theforeman/foreman/pull/5253 for now.

**Pull request closed without merging**